spark/src/main/java/org/apache/iceberg/spark/SparkTableUtil.java

```diff
@@ -151,15 +154,21 @@ private SparkTableUtil() {
    *
    * @param spark a Spark session
    * @param tableIdent a table identifier
+   * @param partitionFilter the partition filter
    * @return all table's partitions
    */
-  public static List<SparkPartition> getPartitions(SparkSession spark, TableIdentifier tableIdent) {
+  public static List<SparkPartition> getPartitions(SparkSession spark, TableIdentifier tableIdent,
```

Comment:
Do we benefit from having this as a Java Optional? Since we have to immediately convert it to Scala maybe we should just pass a normal Map?

Reply:
I think the "filterPartitions" function would just use an empty map as "no filter"?

Reply:
Done